analyzer/runtime/visitors.go

```diff
@@ -221,50 +223,70 @@ func VisitEVMEvent(event *evm.Event, handler *EVMEventHandler) error {
 	switch {
 	case eventMatches(event, evmabi.ERC20.Events["Transfer"]):
 		if handler.ERC20Transfer != nil {
-			if err := handler.ERC20Transfer(
-				evmCommon.SliceEthAddress(event.Topics[1]),
-				evmCommon.SliceEthAddress(event.Topics[2]),
-				event.Data,
+			_, args, err := abiparse.ParseEvent(event.Topics, event.Data, evmabi.ERC20)
+			if err != nil {
+				return fmt.Errorf("parse erc20 transfer: %w", err)
+			}
+			if err = handler.ERC20Transfer(
+				args["_from"].(ethCommon.Address),
+				args["_to"].(ethCommon.Address),
+				args["_value"].(*big.Int),
```

Comment:
See above, I think we should pass in the entire `event` here so the handler can easily reconstruct the `evmLog*` attributes of the event.

Reply:
what was the above? don't wanna reconstruct the evmLog* attributes in this handler. this handler is for doing something based on how the event fits into the ERC-20 system. "parsed" attribute stuff is for how the event fits into the source code of the contract. different things